ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamFilterBuilder.java

```diff
@@ -15,28 +15,27 @@
 
 package io.confluent.ksql.execution.streams;
 
-import io.confluent.ksql.GenericRow;
 import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
 import io.confluent.ksql.execution.context.QueryContext;
 import io.confluent.ksql.execution.context.QueryLoggerUtil;
 import io.confluent.ksql.execution.plan.StreamFilter;
+import io.confluent.ksql.execution.plan.StreamWithKeySerdeFactory;
 import io.confluent.ksql.execution.sqlpredicate.SqlPredicate;
-import org.apache.kafka.streams.kstream.KStream;
 
 public final class StreamFilterBuilder {
   private StreamFilterBuilder() {
   }
 
-  public static <K> KStream<K, GenericRow> build(
-      final KStream<K, GenericRow> kstream,
-      final StreamFilter<KStream<K, GenericRow>> step,
+  public static <K> StreamWithKeySerdeFactory<K> build(
```

Comment:
This is kind of a general point. I notice that we use "build" in the codebase a lot. And often it's used where the builder pattern is not actually being used, rather, something is just being created. I'd suggest only using "build" where there's a real builder pattern in place, and using "create" in the case of a simple factory method.

Reply:
Fair enough. I'll defer this to another PR though to keep the scope of the changes here at a minimum.